my-app/src/components/GenericForm.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { Box } from "@mui/material";
+import FormHelperText from "@mui/material/FormHelperText";
+import Ajv, { JSONSchemaType, Schema } from "ajv";
+import ajvErrors from "ajv-errors";
+
+const ajv = new Ajv({ allErrors: true });
+require("ajv-errors")(ajv);
+
+interface Field {
+  id: string;
+  label: string;
+  type: string;
+  required: boolean;
+  placeholder: string;
+  errorMsg: string;
+  errorMsgVisibility: boolean;
+  options?: { value: string; label: string }[]; // For select fields
+}
+
+interface Props {
+  fields: Field[];
+  customSubmitFunction: (data: Record<string, any>) => void;
+  submitButtonName: string;
+  schema: Schema;
+}
+
+const GenericForm: React.FC<Props> = ({
+  fields,
+  customSubmitFunction,
+  submitButtonName,
+  schema,
+}) => {
+  const [formData, setFormData] = useState<Record<string, string | undefined>>(
+    {}
+  );
+  const [fieldsData, setFieldsData] = useState(fields);
+  const [currentLabelStyle, setCurrentLabelStyle] =
+    useState("signinLabelNormal");
+  const validate = ajv.compile(schema);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    formState: { errors },
+  } = useForm();
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { id, value } = event.target;
+    setFormData({ ...formData, [id]: value });
+  };
+
+  const onChange = () => {
+    setCurrentLabelStyle("signinLabelNormal");
+  };
+  const validateForm = (data: Record<string, any>) => {
+    fields.map((field) => (field.errorMsgVisibility = true)); //Reset Error Messages
```

Comment:
Directly mutating the fields prop to reset error message visibility is not recommended as it modifies the props, which goes against React's unidirectional data flow principle. Consider managing visibility through component state or another approach that doesn't involve mutating props directly.